internal/services/gcp_reservation_service_test.go

```diff
@@ -29,33 +28,65 @@ func TestCreateGCPReservationHandler(t *testing.T) {
 	ctx = stubs.WithReservationDao(ctx)
 	ctx = stubs.WithPubkeyDao(ctx)
 	pk := factories.NewPubkeyRSA()
-	err = stubs.AddPubkey(ctx, pk)
+	err := stubs.AddPubkey(ctx, pk)
 	require.NoError(t, err, "failed to generate pubkey")
 	source, err := Clientstubs.AddSource(ctx, models.ProviderTypeGCP)
 	require.NoError(t, err, "failed to generate GCP source")
 
-	values := map[string]interface{}{
-		"source_id":    source.Id,
-		"image_id":     "80967e7f-efef-4eee-85b0-bd4cef4c455d",
-		"amount":       1,
-		"zone":         "us-central1-a",
-		"machine_type": "n1-standard-1",
-		"pubkey_id":    pk.ID,
-	}
-	if json_data, err = json.Marshal(values); err != nil {
-		t.Fatalf("unable to marshal values to json: %v", err)
-	}
-
-	req, err := http.NewRequestWithContext(ctx, "POST", "/api/provisioning/reservations/gcp", bytes.NewBuffer(json_data))
-	require.NoError(t, err, "failed to create request")
-	req.Header.Add("Content-Type", "application/json")
-
-	rr := httptest.NewRecorder()
-	handler := http.HandlerFunc(services.CreateGCPReservation)
-	handler.ServeHTTP(rr, req)
-
-	require.Equal(t, http.StatusOK, rr.Code, "Handler returned wrong status code")
-
-	stubCount := stubs.GCPReservationStubCount(ctx)
-	assert.Equal(t, 1, stubCount, "Reservation has not been Created through DAO")
+	t.Run("successful reservation", func(t *testing.T) {
+		var err error
+		values := map[string]interface{}{
+			"source_id":    source.Id,
+			"image_id":     "80967e7f-efef-4eee-85b0-bd4cef4c455d",
+			"amount":       1,
+			"zone":         "us-central1-a",
+			"machine_type": "n1-standard-1",
+			"pubkey_id":    pk.ID,
+		}
+		if json_data, err = json.Marshal(values); err != nil {
+			t.Fatalf("unable to marshal values to json: %v", err)
+		}
+
+		req, err := http.NewRequestWithContext(ctx, "POST", "/api/provisioning/reservations/gcp", bytes.NewBuffer(json_data))
+		require.NoError(t, err, "failed to create request")
+		req.Header.Add("Content-Type", "application/json")
+
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(services.CreateGCPReservation)
+		handler.ServeHTTP(rr, req)
+
+		require.Equal(t, http.StatusOK, rr.Code, "Handler returned wrong status code")
+
+		stubCount := stubs.GCPReservationStubCount(ctx)
+		assert.Equal(t, 1, stubCount, "Reservation has not been Created through DAO")
+	})
+
+	t.Run("failed reservation with invalid zone", func(t *testing.T) {
+		var err error
+		require.NoError(t, err, "failed to generate pubkey")
+		source, err := Clientstubs.AddSource(ctx, models.ProviderTypeGCP)
```

Comment:
I do not think there is a need to check for the NoError here, as you declare a new err variable the line before. The err should have the default error value (nil).

Reply:
Was a leftover, great catch :)